takwimu/templates/profile/profile_detail.html

```diff
@@ -55,16 +55,39 @@ <h1>Elections</h1>
         <div class="section-container">
           <section class="clearfix stat-row">
             <h2 class="header-for-columns">Votes Distribution </h2>
-
-            <div class="column-half" id="chart-pie-elections-valid_invalid_dist" data-stat-type="percentage" data-chart-title="Valid Votes Vs Invalid Votes" data-initial-sort="-value"></div>
-            <div class="column-half" id="chart-pie-elections-registered_accred_dist" data-stat-type="percentage" data-chart-title="Registered Voters Vs Total Votes Cast" data-initial-sort="-value"></div>
+            {% if elections.valid_invalid_dist.name != 'No Data Found' %}
+              <div class="column-half" id="chart-pie-elections-valid_invalid_dist" data-stat-type="percentage" data-chart-title="Valid Votes Vs Invalid Votes" data-initial-sort="-value"></div>
+              <div class="column-half" id="chart-pie-elections-registered_accred_dist" data-stat-type="percentage" data-chart-title="Registered Voters Vs Total Votes Cast" data-initial-sort="-value"></div>
+            {% else %}
+              {% include 'profile/no_data_available.html' %}
+            {% endif %}
           </section>
         </div>
         <div class="section-container">
           <section class="clearfix stat-row">
             <h2 class="header-for-columns">Total Votes Per Candidate </h2>
+            {% if elections.valid_invalid_dist.name != 'No Data Found' %}
+              <div class="column-full" id="chart-histogram-elections-candidate_dist" data-stat-type="scaled-percentage" data-chart-title="Percentage Votes Per Candidate" data-initial-sort="-value"></div>
+            {% else %}
+              {% include 'profile/no_data_available.html' %}
+            {% endif %}
+          </section>
+        </div>
+      </article>
 
-            <div class="column-full" id="chart-histogram-elections-candidate_dist" data-stat-type="scaled-percentage" data-chart-title="Percentage Votes Per Candidate" data-initial-sort="-value"></div>
+
+      <article class="clearfix">
+        <header class="section-contents">
+          <h1>Crops</h1>
+        </header>
+        <div class="section-container">
+          <section class="clearfix stat-row">
+            <h2 class="header-for-columns">Crop Production in Tonnes </h2>
+            {% if crops.crop_distribution.name == 'No Data Found' %}
+              {% include 'profile/no_data_available.html' %}
```

Comment:
I think this may generate a lot of info messages. Why not hide the whole chart if no data available?


![too_many_warnings](https://user-images.githubusercontent.com/1779590/43582708-f54e7bb4-9665-11e8-902b-b2f3bd0f4215.png)